### ISSUE TITLE
Mulled test can give incorrect results

### ISSUE THREAD

**Issue by dpryan79** [bioconda/bioconda-utils#601]
There's an unusual edge case where recipes have both a run_test.sh and tests. The run_test.sh is used by conda-build but the tests in the meta.yaml are used during mulled testing. In can occur (e.g., in https://github.com/bioconda/bioconda-recipes/pull/17157, which is a proof of concept) that these commands would have failed if run by conda-build but will pass due to how they're sent to the container. In short, if we have the case of:

    - false
    - false || true

Then the mulled container runs `false && false || true`. Conda correctly fails at the first `false`, since it runs everything in a bash shell script with `set -x`.

**Comment by bgruening** [comment]
So we should change the `&&` to `;`. We should not use run_tests.sh we discourage that, exactly because mulled can not use it currently ... its more for very wired etch cases, where people need to use it. In this case the mata.yml test cases can still be used to test simple stuff in the container.

**Comment by dpryan79** [comment]
Yeah, we either need to use ; or parentheses. This sounds like a good hackathon project for someone :)
